File: 백준/Silver/1946. 신입 사원/신입 사원.py
import sys
t=int(sys.stdin.readline())
for i in range(t):
    n=int(sys.stdin.readline())
    applicants,cnt=[],1
    for i in range(n):
        applicants+=[list(map(int,sys.stdin.readline().split()))]
    applicants.sort(key=lambda x:x[0]);
    cutline=applicants[0][1]
    for i in range(1,n):
        if cutline>applicants[i][1]:
            cutline=applicants[i][1];
            cnt+=1
    print(cnt)

File: 백준/Silver/1946. 신입 사원/신입 사원.c
#include <stdio.h>
#include <stdlib.h>
int compare(const void *pa, const void *pb) {
    const int *a = *(int **)pa;
    const int *b = *(int **)pb;
    return a[0] > b[0];
}
int main(void) {
    int t, n, cnt, cutline, **applicants;
    scanf("%d", &t);
    for (int i = 0; i < t; i++) {
        scanf("%d", &n);
        applicants = (int **)malloc(sizeof(int *) * n);
        for (int j = 0; j < n; j++) {
            scanf("%d %d", &cnt, &cutline);
            applicants[j] = (int *)malloc(sizeof(int) * 2);
            applicants[j][0] = cnt;
            applicants[j][1] = cutline;
        }
        qsort(applicants, n, sizeof(int *), compare);
        cnt = 1;
        cutline = applicants[0][1];
        for (int j = 1; j < n; j++) {
            if (cutline > applicants[j][1]) {
                cutline = applicants[j][1];
                cnt += 1;
            }
        }
        printf("%d\n", cnt);
    }
    return 0;
}